src/js/stores/MesosStateStore.js

```diff
@@ -90,22 +90,36 @@ class MesosStateStore extends GetSetBaseStore {
       .subscribe(this.onStreamData, this.onStreamError);
   }
 
+  getLastMesosState() {
+    const defaultMesosState = {
+      executors: [],
+      frameworks: [],
+      master_info: [],
+      slaves: [],
+      tasks: []
+    };
+
+    const lastState = this.get("lastMesosState") || {};
+
+    return Object.assign({}, defaultMesosState, lastState);
```

Comment:
still dont understand why we arent just writing:
```
return Object.assign({
  executors: [],
  frameworks: [],
  master_info: [],
  slaves: [],
  tasks: []
}, this.get("lastMesosState"));
```

Reply:
That is also my opinion of a more robust and future-proof way to solve this issue.

Reply:
FTFY 👍